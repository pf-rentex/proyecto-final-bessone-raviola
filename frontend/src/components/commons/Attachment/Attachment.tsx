import React from 'react';
import { HiDownload, IoClose } from 'react-icons/all';

interface IAttachmentProps {
    name: string;
    size: string;
    attachedDate: string;
}

const Attachment = ({ name, size, attachedDate }: IAttachmentProps) => {
    return (
        <div className="bg-blue-900 py-5 md:py-2 px-10 rounded shadow-xl">
            <div className="flex flex-col space-y-3 md:space-y-0 md:flex-row items-center w-full space-x-5">
                <div className="flex flex-col justify-center space-y-3">
                    <p className="text-white">{name}</p>
                    <div className="flex space-x-5 md:space-x-10 justify-center md:justify-start">
                        <p className="text-blue-300 font-thin">{size}</p>
                        <p className="text-blue-300 font-thin">
                            {attachedDate}
                        </p>
                    </div>
                </div>
                <div className="flex w-full space-x-5 justify-center md:justify-end">
                    <div className="rounded-full bg-primary p-2 shadow-xl">
                        <HiDownload className="text-white" />
                    </div>
                    <div className="rounded-full bg-red-600 p-2 shadow-xl">
                        <IoClose className="text-white" />
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Attachment;
