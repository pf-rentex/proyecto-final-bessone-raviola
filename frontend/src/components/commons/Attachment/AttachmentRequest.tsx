import React from 'react';
import { AiOutlineFileText } from 'react-icons/all';

interface IAttachmentRequestProps {
    handleFile: Function;
}

const AttachmentRequest = ({ handleFile }: IAttachmentRequestProps) => {
    const hiddenFileInput = React.useRef(null);
    const handleClick = () => {
        let fileInput: any = hiddenFileInput.current;
        fileInput.click();
    };
    const handleChange = (event: any) => {
        let fileUploaded = event.target.files[0];
        handleFile(fileUploaded);
    };
    return (
        <div className="bg-blue-700 h-20 py-2 px-10 border-dashed border-2 border-blue-900 rounded shadow-xl flex items-center cursor-pointer">
            <input
                type="file"
                className="hidden"
                ref={hiddenFileInput}
                onChange={handleChange}
            />
            <div
                className="flex items-center w-full justify-between lg:space-x-5"
                onClick={handleClick}
            >
                <p className="text-white">Haz click para adjuntar archivo</p>
                <AiOutlineFileText className="text-blue-300 w-8 h-8" />
            </div>
        </div>
    );
};

export default AttachmentRequest;
