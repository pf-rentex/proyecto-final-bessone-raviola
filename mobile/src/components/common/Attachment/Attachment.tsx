import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableHighlight,
  Dimensions,
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from 'react-native-responsive-screen';

interface IAttachmentProps {
  name: string;
  size: string;
  date?: string;
  handleDelete: Function;
}

const Attachment = ({name, size, date, handleDelete}: IAttachmentProps) => {
  const styles = StyleSheet.create({
    container: {
      backgroundColor: '#1E4663',
      height: hp('8%'),
      borderRadius: 6,
      flex: 1,
      shadowColor: '#000',
      shadowOffset: {
        width: 0,
        height: 8,
      },
      shadowOpacity: 0.44,
      shadowRadius: 10.32,
      elevation: 16,
      padding: 15,
      flexDirection: 'row',
      alignItems: 'center',
      marginVertical: hp('2%'),
    },
    deleteButton: {
      borderRadius:
        Math.round(
          Dimensions.get('window').width + Dimensions.get('window').height,
        ) / 2,
      width: wp(8),
      height: wp(8),
      backgroundColor: '#EA4D4D',
      justifyContent: 'center',
      alignItems: 'center',
      marginHorizontal: wp(1),
    },
    downloadButton: {
      borderRadius:
        Math.round(
          Dimensions.get('window').width + Dimensions.get('window').height,
        ) / 2,
      width: wp(8),
      height: wp(8),
      backgroundColor: '#6DE4FE',
      justifyContent: 'center',
      alignItems: 'center',
      marginHorizontal: wp(1),
    },
  });
  return (
    <View style={styles.container}>
      <View style={{flex: 1}}>
        <View style={{flexDirection: 'column'}}>
          <Text style={{color: 'white', fontWeight: 'bold'}}>{name}</Text>
          <View style={{flexDirection: 'row'}}>
            <Text style={{color: '#6DE4FE', marginRight: wp(5)}}>{size}</Text>
            <Text style={{color: '#6DE4FE'}}>{date}</Text>
          </View>
        </View>
      </View>
      <View
        style={{
          alignSelf: 'center',
          flexDirection: 'row',
        }}>
        <TouchableHighlight style={styles.downloadButton} onPress={() => {}}>
          <Icon name='download' size={20} color='white' />
        </TouchableHighlight>
        <TouchableHighlight style={styles.deleteButton} onPress={handleDelete}>
          <Icon name='close' size={20} color='white' />
        </TouchableHighlight>
      </View>
    </View>
  );
};

export default Attachment;
