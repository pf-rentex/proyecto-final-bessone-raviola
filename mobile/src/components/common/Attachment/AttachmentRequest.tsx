import React from 'react';
import {View, Text, StyleSheet, TouchableHighlight} from 'react-native';
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from 'react-native-responsive-screen';
import {useNavigation} from '@react-navigation/native';

const AttachmentRequest = ({fileType}: any) => {
  const navigation = useNavigation();
  const styles = StyleSheet.create({
    container: {
      backgroundColor: 'rgba(30,70,99,0.5)',
      height: hp('10%'),
      borderRadius: 6,
      borderStyle: 'dashed',
      borderWidth: 1,
      shadowColor: '#000',
      shadowOffset: {
        width: 0,
        height: 8,
      },
      shadowOpacity: 0.44,
      shadowRadius: 10.32,
      padding: 15,
      justifyContent: 'center',
      alignItems: 'center',
    },
  });

  return (
    <View style={styles.container}>
      <TouchableHighlight
        onPress={() => {
          navigation.navigate('DocumentViewer', {fileType});
        }}>
        <Text style={{color: 'white', fontWeight: 'bold'}}>
          Adjuntar o Escanear archivo
        </Text>
      </TouchableHighlight>
    </View>
  );
};

export default AttachmentRequest;
