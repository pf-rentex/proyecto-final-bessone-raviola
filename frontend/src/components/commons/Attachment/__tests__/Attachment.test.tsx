import React from 'react';
import { getByText, render } from '@testing-library/react';
import Attachment from '../Attachment';

describe('Attachment', () => {
    test('renders correctly', () => {
        const date = new Date(2020, 4, 12);
        const { container, getByText } = render(
            <Attachment size="10kb" attachedDate={date.toString()} handleDelete={() => {}} name="Attachment" />,
        );

        getByText('Attachment');
        getByText('10kb');
        getByText(date.toString());
        expect(container).toMatchSnapshot();
    });
});
