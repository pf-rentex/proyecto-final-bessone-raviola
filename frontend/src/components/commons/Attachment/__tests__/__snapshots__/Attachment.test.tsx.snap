// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachment renders correctly 1`] = `
<div>
  <div
    class="bg-blue-900 py-5 md:py-2 px-10 rounded shadow-xl"
  >
    <div
      class="flex flex-col space-y-3 md:space-y-0 md:flex-row items-center w-full space-x-5"
    >
      <div
        class="flex flex-col justify-center space-y-3"
      >
        <p
          class="text-white"
        >
          Attachment
        </p>
        <div
          class="flex space-x-5 md:space-x-10 justify-center md:justify-start"
        >
          <p
            class="text-blue-300 font-thin"
          >
            10kb
          </p>
          <p
            class="text-blue-300 font-thin"
          >
            Tue May 12 2020 00:00:00 GMT-0300 (Argentina Standard Time)
          </p>
        </div>
      </div>
      <div
        class="flex w-full space-x-5 justify-center md:justify-end"
      >
        <div
          class="rounded-full bg-primary p-2 shadow-xl cursor-pointer"
        >
          <svg
            class="text-white"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3 17a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zm3.293-7.707a1 1 0 011.414 0L9 10.586V3a1 1 0 112 0v7.586l1.293-1.293a1 1 0 111.414 1.414l-3 3a1 1 0 01-1.414 0l-3-3a1 1 0 010-1.414z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="rounded-full bg-red-600 p-2 shadow-xl cursor-pointer"
        >
          <svg
            class="text-white"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
